### PR TITLE
fix(@desktop/Profile): validate social links before saving #7926

### PR DESCRIPTION
**Pull request by mprakhov**
### What does the PR do

Validate input social links via regexp on save

### Affected areas

Profile social links saving

### CODE REVIEW

File: src/app_service/common/utils.nim [hunk 1]
```diff
@@ -51,3 +51,11 @@ proc defaultDataDir(): string =
       else:
         targetDir
   absolutePath(joinPath(parentDir, "Status"))
+
+proc validateLink*(link: string): bool =
+  result = true
+  if link.len() != 0:
+    if not match(
+        link, re"[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)", 0):
```
Comment (caybro) [inline src/app_service/common/utils.nim:59]: http URLs can also include UTF-8 text

**Review by caybro — CHANGES_REQUESTED** [review]

**Review by borismelnik — APPROVED** [review]
LGTM, just fix @caybro comment :)

File: src/app_service/common/utils.nim (hunk 1 above)
Comment (caybro) [reply src/app_service/common/utils.nim:59]: http://ko.wikipedia.org/wiki/%EC%9C%84%ED%82%A4%EB%B0%B1%EA%B3%BC:%EB%8C%80%EB%AC%B8

Comment (caybro) [reply src/app_service/common/utils.nim:59]: My bad, they do show up as Unicode but must be percent encoded on input

**Review by caybro — APPROVED** [review]

**Pull request merged**